### DISCUSSION TITLE
Cara pengiriman MedicationRequest

### DISCUSSION THREAD

**Discussion by AlulCode45** [ivanwilliammd/satusehat-integration#116] [Q&A (Tanya Jawab)] [answered]
Ketika saya melakukan request Medication request ke satusehat. dan pada kondisi ini saya meenggunakan 
```
src/FHIR/MedicationRequest.php
```
dan di sini saya mengirimkan menggunakan kode di bawah ini
```php
$medication = new MedicationRequest();
$medication->addPrescriptionIdentifier("RSP-".Str::random(8));
$medication->addPrescriptionIdentifier("RSP-".Str::random(8));

$medication->setStatus();
$medication->setIntent();
$medication->addCategory();
$medication->setSubject($pasienData->patient_id, $pasienData->nama_pasien);
$medication->setEncounter($pasien->uuid);
$medication->setRequester($dokter->ihs, $dokter->name);
$medication->addNote($data['aturan_tambahan']);
$medication->setMedicationReference($obat->reference_id, $obat->name);
$medication->addDosageInstruction(
   intval($data['frekuensi_penggunaan_obat_per_hari']),
   $data['rute'],
   $data['timing'],
   $data['aturan_tambahan']
);
$medication->setSubstitution();

$medication->json();
[$responseCode, $response] = $medication->post();

return [$responseCode, $response];
```

dan mendapatkan error seperti ini
```
array:2 [▼ // app\Models\ResepModels.php:37
  0 => 400
  1 => {#3216 ▼
    +"resourceType": "OperationOutcome"
    +"text": {#3221 ▶}
    +"issue": array:3 [▼
      0 => {#2899 ▼
        +"severity": "error"
        +"code": "value"
        +"details": {#3209 ▼
          +"text": "medicationReference has to refer to Medication (Rule Number: 10135)"
        }
        +"expression": array:1 [▼
          0 => "MedicationRequest.medicationReference"
        ]
      }
      1 => {#3202 ▼
        +"severity": "error"
        +"code": "value"
        +"details": {#3219 ▼
          +"text": "Wrong reference format: bcac4b8e-a76f-4132-af2a-ef2a542b6455 (RuleNumber: 10135)"
        }
        +"expression": array:1 [▼
          0 => "MedicationRequest.medicationReference"
        ]
      }
      2 => {#3184 ▼
        +"severity": "error"
        +"code": "value"
        +"details": {#3155 ▼
          +"text": "Element not found: MedicationRequest.dispenseRequest (RuleNumber: 10453)"
        }
        +"expression": array:1 [▼
          0 => "MedicationRequest.dispenseRequest"
        ]
      }
    ]
  }
]
```

**Comment by AlulCode45** [comment]
![image](https://github.com/user-attachments/assets/b937e994-d963-4131-b2c3-595ffefe1159)
Menurut analisis saya ini dikarenakan lbrary  pada function **setMedicationReference**, tidak menggunakan prefix Medication sebelum reference obatnya

Jika saya menggunakan ini maka tidak ada error lagi.
```php

//Sebelum
 public function setMedicationReference($reference, $display)
    {
        $this->medication_request['medicationReference']['reference'] = $reference;
        $this->medication_request['medicationReference']['display'] = $display;
    }
// Sesudah saya perbaiki
 public function setMedicationReference($reference, $display)
    {
        $this->medication_request['medicationReference']['reference'] = 'Medication/'.$reference;
        $this->medication_request['medicationReference']['display'] = $display;
    }
```

Sebenarnya sudah saya ajukan pull request, namun @ivanwilliammd memperbaiki yang observation saja
[https://github.com/ivanwilliammd/satusehat-integration/pull/114](https://github.com/ivanwilliammd/satusehat-integration/pull/114)

**Reply by ivanwilliammd** [reply, marked as answer]
Sayang sekali belum lengkap cara pandangnya

Coba lihat resource Medication dan pastikan kembali di MedicationRequest.contained sudah terdapat object class Medication

Setelah dicoba baru kita diskusikan kembali. Tks

**Reply by AlulCode45** [reply]
Baik bisa di mengerti, berarti kesalahan saya tidak menggunakan fungsi addContainer

**Reply by AlulCode45** [reply]
![image](https://github.com/user-attachments/assets/75881f80-3ad9-465f-851f-bac6edbcd752)
![image](https://github.com/user-attachments/assets/6923990c-150b-4a7e-a5ec-eb218f3f5671)
Terkait contained dan juga noncontained, nah berarti sebenarnya ada 2 cara untuk melakukan pengiriman obat ya dok?

**Reply by ivanwilliammd** [reply]
Kesimpulan dan ini sudah tested ya..

Buat objek Medication dahulu, beserta dummy Medication.id (misal 123456789)
Buat object MedicationRequest --> tambahkan contained dengan Medication object class
MedicationRequest.medication.reference = #123456789

**Comment by ivanwilliammd** [comment]
Kesimpulan dan ini sudah tested ya..

1. Buat objek Medication dahulu, beserta dummy Medication.id (misal 123456789)
2. Buat object MedicationRequest --> tambahkan **contained** dengan Medication object class
3. MedicationRequest.medication.reference = #123456789
